test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -397,6 +398,20 @@ def closure():
         # Load the optimizer state dict, check that no exception is raised
         optimizer.load_state_dict(optimizer_state_dict)
 
+        # Check that the states are not None, but {}
+        for state in optimizer.state.values():
+            for _, _ in state.items():
+                pass
+
+        # Test the state dict materialization on all ranks
```

Comment:
testing the new option introduced by this PR, all the ranks get the full state dict